yarn-project/circuits.js/src/cbind/constants.in.ts

```diff
@@ -0,0 +1,59 @@
+import * as fs from 'fs';
+import { dirname, join } from 'path';
+import { fileURLToPath } from 'url';
+
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const CPP_FILE_PATH = join(__dirname, '../../../../circuits/cpp/src/aztec3/constants.hpp');
+const TS_FILE_PATH = join(__dirname, '/constants.gen.ts');
+
+/**
+ * Convert the C++ constants to TypeScript.
+ */
+function convertToTypeScript(): void {
+  let content: string = fs.readFileSync(CPP_FILE_PATH, 'utf-8');
+
```

Comment:
Hmm, why this manual parsing approach? I was expecting that we'd just leverage something like msgpack to write out a json of all the constants, read them, then make the file based on them

Reply:
> Hmm, why this manual parsing approach? 
I told chatgpt to write the script and it worked.

> Also if we wanted to use these constants as constants in types, we can't do log2 as this decays the type to number (instead of e.g. the type 5)
Good point. Will try refactoring it with msgpack.